packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -31,31 +36,61 @@ const defaultProps = {
   component: 'a',
   isDisabled: false,
   href: '#',
-  onClick: Function.prototype
+  onClick: Function.prototype,
+  index: -1,
+  context: {
+    keyHandler: Function.prototype,
+    sendRef: Function.prototype
+  }
 };
 
 class DropdownItem extends React.Component {
+  componentWillMount() {
+    this.ref = React.createRef();
+  }
+
+  componentDidMount() {
+    this.props.context.sendRef(this.props.index, this.ref, this.props.isDisabled);
+  }
+
+  onKeyDown = event => {
+    // Detected key press on this item, notify the menu parent so that the appropriate
+    // item can be focused
+    if (event.key === 'Tab') return;
+    event.preventDefault();
+    if (event.key === 'ArrowUp') {
+      this.props.context.keyHandler(this.props.index, 'up');
+    } else if (event.key === 'ArrowDown') {
+      this.props.context.keyHandler(this.props.index, 'down');
+    } else if (event.key === 'Enter') {
+      this.ref.click();
```

Comment:
This, i beleive should be `this.ref.current.click()`. Also since it is possible to pass a component in you will need to make sure that `click` exists.